x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/GrammarTests.java

```diff
@@ -28,20 +29,38 @@
  */
 public class GrammarTests extends ESTestCase {
 
-    public void testGrammar() throws Exception {
+    public void testValidQueries() throws Exception {
         EqlParser parser = new EqlParser();
-        List<Tuple<String, Integer>> lines = readQueries("/grammar-queries.eql");
+        List<Tuple<String, Integer>> lines = readQueries("/valid-queries.eql");
         for (Tuple<String, Integer> line : lines) {
             String q = line.v1();
             try {
                 parser.createStatement(q);
             } catch (ParsingException pe) {
                 if (pe.getErrorMessage().startsWith("Does not know how to handle")) {
                     // ignore for now
+                } else {
+                    throw new ParsingException(new Source(pe.getLineNumber() + line.v2() - 1, pe.getColumnNumber(), q),
+                        pe.getErrorMessage() + " inside statement <{}>", q);
                 }
-                else {
+            }
+        }
+    }
+    public void testUnsupportedQueries() throws Exception {
+        EqlParser parser = new EqlParser();
+        List<Tuple<String, Integer>> lines = readQueries("/unsupported-queries.eql");
+        for (Tuple<String, Integer> line : lines) {
+            String q = line.v1();
```

Comment:
Use `expectedThrows` instead (see the `EsTestCase`) which does the try/catch plus validation of the exception type.